### PR TITLE
Creating basic home page

### PR DESCRIPTION
**Pull request by AvitalGitelman**

### CODE REVIEW

File: eatin-client/src/homePage/CategoryCarousel.tsx [hunk 1]
```diff
@@ -0,0 +1,177 @@
+import { FC } from "react";
+import { Button, Rating } from "@mui/material";
+import CancelIcon from "@mui/icons-material/Cancel";
+import { Carousel, CarouselItem, CarouselProps } from "../components/ui/Carousel";
+import { useNavigate } from "react-router-dom";
+
+export class Recipe {
```
Comment (edenshaul) [inline eatin-client/src/homePage/CategoryCarousel.tsx:7]: I don't think that you should declare all of the params as `string`, some of them are `int` and so on

File: eatin-client/src/graphql/mutations/recipes.mutation.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import {gql} from '@apollo/client';
+
+export const CREATE_RECIPE = gql`
+    mutation createRecipe($recipe_title: String!,
```
Comment (edenshaul) [inline eatin-client/src/graphql/mutations/recipes.mutation.ts:4]: why is everything here `string`?

File: eatin-client/src/homePage/CategoryCarousel.tsx (hunk 1 above)
Comment (edenshaul) [inline eatin-client/src/homePage/CategoryCarousel.tsx:7]: also why is this in the same file as the `CategoryCarousel`? I think you should separate these classes to different files in different folders

File: eatin-client/src/homePage/HomePage.tsx [hunk 3]
```diff
@@ -1,5 +1,176 @@
...
+                        recipe.difficulty,
+                    ),
+                ),
+        );
+        return items;
+    }
+
+    // This is temp
+    useEffect(() => {
+        const eggItems = getItemsArray(eggs);
```
Comment (edenshaul) [inline eatin-client/src/homePage/HomePage.tsx:94]: are you fixing this in your next issue? can you open an issue in our board for this
also change for `TODO` so we won't miss any comments

File: eatin-client/src/homePage/HomePage.tsx [hunk 4]
```diff
@@ -1,5 +1,176 @@
...
+        },
+    ];
+
+    const updateSearchResult = () => {
+        setSearchResult(filterSearchVal);
+        setfilterSearchVal("");
+    };
+
+    const showMyRecipes = () => {
+        // here we will update the shown recipes....
```
Comment (edenshaul) [inline eatin-client/src/homePage/HomePage.tsx:134]: `// TODO ...`

File: eatin-client/src/homePage/HomePage.tsx [hunk 5]
```diff
@@ -1,5 +1,176 @@
...
+    };
+
+    const showSavedRecipes = () => {
+        // here we will update the shown recipes....
```
Comment (edenshaul) [inline eatin-client/src/homePage/HomePage.tsx:138]: `// TODO ...`

File: eatin-client/src/homePage/HomePage.tsx [hunk 6]
```diff
@@ -1,5 +1,176 @@
...
+                    My Recipes
+                </Button>
+                <Button className="recipesBtn" onClick={showSavedRecipes}>
+                    Saved Recipes
+                </Button>
+            </div>
+
+            {!searchFilter ? <RecommentedFeed currentRecipes={currentShownRecipes} /> : <div></div>}
+
+            <div></div>
```
Comment (edenshaul) [inline eatin-client/src/homePage/HomePage.tsx:173]: del this

File: eatin-client/src/homePage/HomePage.tsx [hunk 7]
```diff
@@ -1,5 +1,176 @@
...
+            { name: Cuisine.Greek.toString(), items: greekItems },
+        ]);
+    }, [currentShownRecipes, cakes, eggs, japanese, greek]);
+
+    const currentFilterOptions: {
+        name: string;
+        options: string[];
+        funcToUpdate: (arg0: string) => void;
+    }[] = [
+        {
```
Comment (edenshaul) [inline eatin-client/src/homePage/HomePage.tsx:111]: do this as a query from the backend, or if not now for the prototype - add a `TODO` comment and open an issue in the board

File: eatin-client/src/index.css [hunk 8]
```diff
@@ -0,0 +1,26 @@
+body {
+  margin: 0;
+  width: 100%;
+  height: 100%;
+  /* overflow: hidden; */
```
Comment (edenshaul) [inline eatin-client/src/index.css:5]: del

File: eatin-client/src/generated/graphql.tsx [hunk 9]
```diff
@@ -70,6 +155,32 @@ export type CreateUserMutationVariables = Exact<{
...
+
+
+export type GetTopRatedRecipesByCategoryQuery = { __typename?: 'Query', topRecipesByCategory: Array<{ __typename?: 'Recipes', index: string, recipe_title: string, url: string, record_health: string, vote_count: string, rating: string, description: string, cuisine: string, course: string, diet: string, prep_time: string, cook_time: string, ingredients: string, instructions: string, author: string, tags: string, category: string, image: string, difficulty: string }> };
+
+export type GetTopRatedRecipesByCuisineQueryVariables = Exact<{
+  cuisine: Scalars['String'];
+}>;
+
+
+export type GetTopRatedRecipesByCuisineQuery = { __typename?: 'Query', topRecipesByCuisine: Array<{ __typename?: 'Recipes', index: string, recipe_title: string, url: string, record_health: string, vote_count: string, rating: string, description: string, cuisine: string, course: string, diet: string, prep_time: string, cook_time: string, ingredients: string, instructions: string, author: string, tags: string, category: string, image: string, difficulty: string }> };
```
Comment (Shirley-Chesner) [inline eatin-client/src/generated/graphql.tsx:182]: repetition of the same recipe type, use the Recipe type you created in Types file.

do something like this: Array<Recipe & {_typename?: 'Recipes' }>.
you can define this type at the same file.

Comment (Shirley-Chesner) [reply eatin-client/src/generated/graphql.tsx:182]: also the file name extension should be .ts and not .tsx, we use tsx only for react components

**Pull request merged**